portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java

```diff
@@ -34,13 +34,20 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query();
-
+    String commentId = request.getParameter("comment-id");
+    Query query = new Query("Comment");
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
-
-    for (Entity entity : results.asIterable()) {
+    if (commentId.equals("all")) {
```

Comment:
It's best to avoid using "special" IDs like this. Instead, it would be better to have a separate parameter for this API that can specify whether you want to delete all comments or only some subset by ID.